libarchive/archive_read_support_format_7zip.c

```diff
@@ -518,6 +519,9 @@ archive_read_format_7zip_bid(struct archive_read *a, int best_bid)
 	if (memcmp(p, _7ZIP_SIGNATURE, 6) == 0)
 		return (48);
 
+	if (memcmp(p, _MSI_SIGNATURE, 8) == 0)
+		return (48);
```

Comment:
This seems to be at odds with your description. Bidding 48 will ensure that the 7zip format *is* selected for all MSI archives.